src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -955,7 +955,19 @@ public SolutionState RemoveProjectReference(ProjectId projectId, ProjectReferenc
             var oldProject = this.GetProjectState(projectId)!;
             var newProject = oldProject.RemoveProjectReference(projectReference);
 
-            return this.ForkProject(newProject, newDependencyGraph: _dependencyGraph.WithProjectReferences(projectId, newProject.ProjectReferences.Select(p => p.ProjectId)));
+            ProjectDependencyGraph newDependencyGraph;
+            if (newProject.ContainsReferenceToProject(projectReference.ProjectId))
```

Comment:
so i have to admit this reads super weird.  3 lines above we have `var newProject = oldProject.RemoveProjectReference(projectReference);`  Then, right here, we ask if `newProject` contains a reference to `projectReference` (even though we got `newProject` by removing such a reference.

It feels like something is either incorrect, or this is a op that is necessary for correctness that isn't handled by RemoveProjectReference.  In either event, doc'in this in the code would be highly valuable IMO.

Reply:
n/m, i see this doc'ed below.  it's just still super weird.  i'm surprised we allow this to be a state the system is allowed to get into.

Reply:
The comment is inside the block.